### DISCUSSION TITLE
Same solcx issue

### DISCUSSION THREAD

**Discussion by shlok-codes** [smartcontractkit/full-blockchain-solidity-course-py#614] [Q&A] [answered]
Hi team,

Same issue as I've seen posted here often about solcx. 

My specific error is: 

File "deploy.py", line 1, in <module>
    from solcx import compile_standard, install_solc
ImportError: No module named solcx

After many tries I resorted to just copy and pasted the latest code from the course repo:

```Solidity
from solcx import compile_standard, install_solc

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

# We add these two lines that we forgot from the video!
print("Installing...")
install_solc("0.6.0")

# Solidity source code
compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {
                    "*": ["abi", "metadata", "evm.bytecode", "evm.bytecode.sourceMap"]
                }
            }
        },
    },
    solc_version="0.6.0",
)
print(compiled_sol)

```
I have installed solcx correctly: pip3 install py-solc-x

Have installed/unistalled many times too.  Please advise. I am sure this is very basic. But have been stuck on this error for 3 hours.

**Comment by shlok-codes** [comment, marked as answer]
Have this issue with web3 as well.

I have been searching everywhere for what the issue may be (discussions on this course, google, stackoverflow). And the only thing that worked for me is to create a virtual environment and then install web3 and solcx there.

I hope this is good enough to continue with this course. 

I would however love to figure out why I can't seem to get these modules to work after I install them!

**Reply by cromewar** [reply]
Yes is important to use a different python virtual env for each project to avoid conflicts between packages.

**Reply by jt-berg-whitt** [reply]
what does this mean? how do you install something in a vir env?

**Reply by cromewar** [reply]
@jwhitty1  refer to this: https://virtualenv.pypa.io/en/latest/

**Reply by jt-berg-whitt** [reply]
that doesnt answer my question at all

**Comment by spencergoff** [comment]
I resolved this issue by using the command `python3 deploy.py ` instead of `python deploy.py `.
